packages/rsocket-core/src/RSocketClient.js

```diff
@@ -214,11 +214,17 @@ class RSocketClientSocket<D, M> implements ReactiveSocket<D, M> {
       dataMimeType,
       keepAlive,
       lifetime,
-      metadata,
       metadataMimeType,
-      data,
+      payload,
     } = config.setup;
 
+    const serializers = config.serializers || (IdentitySerializers: any);
+    const data = payload !== undefined
+      ? serializers.data.serialize(payload.data)
+      : undefined;
+    const metadata = payload !== undefined
+      ? serializers.metadata.serialize(payload.metadata)
+      : undefined;
```

Comment:
It seems you don't need to check for `undefined`, the serializers just return `undefined` when you pass `undefined`

Reply:
In that case, I'm checking the payload on "nullability" in order to avoid subsequently thrown `Cannot read property 'metadata' of undefined`

Reply:
You're right, sorry I read it too quickly.